interpreter/src/test/scala/pl/touk/nussknacker/engine/compile/ValidationsSpec.scala

```diff
@@ -0,0 +1,122 @@
+package pl.touk.nussknacker.engine.compile
+
+import cats.data.NonEmptyList
+import cats.data.Validated.{Invalid, Valid}
+import org.scalatest.Inside
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.should.Matchers
+import pl.touk.nussknacker.engine.api._
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.EmptyMandatoryParameter
+import pl.touk.nussknacker.engine.api.definition.MandatoryParameterValidator
+import pl.touk.nussknacker.engine.definition.DefinitionExtractor.ObjectDefinition
+import pl.touk.nussknacker.engine.definition.SubprocessDefinitionExtractor
+import pl.touk.nussknacker.engine.graph.evaluatedparam
+import pl.touk.nussknacker.engine.graph.expression.Expression
+
+
+class ValidationsSpec extends AnyFunSuite with Matchers with Inside {
+  test("should validate fragment: params definitions with actual params") {
+    val subprocessDefinitionExctractor = getSubprocessDefinitionExtractor
+
+    val nodeId = NodeId("fragment1")
+    val fragmentDefinitionParams = subprocessDefinitionExctractor.extractBySubprocessId(nodeId.id)
+    val fragmentActualParams: List[evaluatedparam.Parameter] = List(
+        evaluatedparam.Parameter("P1", Expression("spel", "'ala'")),
+        evaluatedparam.Parameter("P2", Expression("spel", "123")),
+        evaluatedparam.Parameter("P3", Expression("spel", "'ma'"))
+    )
+
+    val validationResult = Validations.validateParametersSafe(fragmentDefinitionParams, fragmentActualParams)(nodeId)
```

Comment:
I don't like the level which you testing. You test internal implementation instead of what user see. What if we change the way how this subprocess validations works? e.g. change validateParametersSafe which BTW IMO is not a way to how - I left the comment there. Take a look how it is done in `NodeDataValidatorSpec` - it abstracts from how internals are realized - only test what user see.